src/BloomBrowserUI/bookEdit/js/synphonyApi.js

```diff
@@ -44,78 +38,64 @@ var SynphonyApi = (function () {
             }
         }
     };
-
     SynphonyApi.prototype.loadFromLangData = function (langData) {
         if (!this.source)
             this.source = new ReaderSettings();
-
         if (this.source.letters === '') {
             var sorted = langData.LanguageSortOrder.join(' ').toLowerCase().split(' ');
             sorted = _.uniq(sorted);
-
             this.source.letters = sorted.join(' ');
         }
     };
-
     SynphonyApi.fireCSharpEvent = function (eventName, eventData) {
-        //noinspection TaskProblemsInspection
-        var event = new MessageEvent(eventName, { 'view': window, 'bubbles': true, 'cancelable': true, 'data': eventData });
-        document.dispatchEvent(event);
+        fireCSharpAccordionEvent(eventName, eventData);
```

Comment:
These 'fireCSharpEvent' methods appear to be scattered all over. Would it be any better if I used the one in bloomEditing.js? Unless someone can figure out how to get Typescript to accept this version of a MessageEvent ctor, we need the version of the method that we use to be in un-generated javascript.